lang/py/src/avro/io.py

```diff
@@ -341,6 +389,71 @@ def write_double(self, datum):
     self.write(chr((bits >> 48) & 0xFF))
     self.write(chr((bits >> 56) & 0xFF))
 
+  def write_decimal_bytes(self, datum, scale):
+    """
+    Decimal in bytes are encoded as long. Since size of packed value in bytes for
+    signed long is 8, 8 bytes are written.
+    """
+    sign, digits, exp = datum.as_tuple()
+    if exp > scale:
+      raise AvroTypeException('Scale provided in schema does not match the decimal')
+
+    unscaled_datum = 0
+    for digit in digits:
+      unscaled_datum = (unscaled_datum * 10) + digit
+
+    bits_req = unscaled_datum.bit_length()
+    if sign:
```

Comment:
The logic for handling the number of bytes used doesn't look correct. For example, `255` gets encoded as a single byte (when it should require two). I think you also need to make sure positive numbers have a `0` in the left-most bit (e.g. by removing line 407 and tweaking lines 410-414).